sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
Heyoeo. From my point of view, the story is seems simple: there is a user who might want to use this timer for some use-case. Now there are two ways to go about it:

1. If the rough consensus is "don't use xtimer for that use-case", then it's the end of the story, and we should document somewhere that it is not recommended to use xtimer in this kind of use-case. 

2. If the consensus is "maybe we could this timer in this use-case" then we need to think about how to accommodate the needs of the user. 

If so, reading the above, I think we are in case 1. Fair summary? Or did I miss something?